File: src/main/java/com/hrms/hr_payroll_management_system/service/PositionService.java
package com.hrms.hr_payroll_management_system.service;

import com.hrms.hr_payroll_management_system.dto.request.position.AssignPositionOrganizationRequest;
import com.hrms.hr_payroll_management_system.dto.request.position.CreatePositionRequest;
import com.hrms.hr_payroll_management_system.dto.request.position.UpdatePositionRequest;
import com.hrms.hr_payroll_management_system.dto.response.position.PositionResponse;

import java.util.List;

public interface PositionService {

    PositionResponse create(CreatePositionRequest request);

    List<PositionResponse> getAll();

    PositionResponse getById(Long id);

    PositionResponse update(
            Long id,
            UpdatePositionRequest request
    );

    void delete(Long id);

    PositionResponse assignOrganization(
        Long positionId,
        AssignPositionOrganizationRequest request
    );

    List<PositionResponse> getByCompanyId(
            Long companyId
    );

    List<PositionResponse> getByBranchId(
            Long branchId
    );

    List<PositionResponse> getByDepartmentId(
            Long departmentId
    );
}

File: frontend/src/services/position.service.js
import { positionApi } from '../api/position.api';

export const positionService = {
  async list() {
    const res = await positionApi.getAll();
    return res.data || [];
  },

  async create({ name, description, companyId, branchId, departmentId, level }) {
    const created = await positionApi.create({ name, description });
    if (companyId && branchId && departmentId && level) {
      const res = await positionApi.assignOrganization(created.data.id, { companyId, branchId, departmentId, level });
      return res.data;
    }
    return created.data;
  },

  async update(id, { name, description, companyId, branchId, departmentId, level }) {
    await positionApi.update(id, { name, description });
    if (companyId && branchId && departmentId && level) {
      const res = await positionApi.assignOrganization(id, { companyId, branchId, departmentId, level });
      return res.data;
    }
    const res = await positionApi.getById(id);
    return res.data;
  },

  async remove(id) {
    await positionApi.remove(id);
  },
};